### PR TITLE
Add "Switch site" page

### PR DESCRIPTION
**Pull request by dsas**
Part of https://github.com/Automattic/wp-calypso/issues/88229
Required by https://github.com/Automattic/wp-calypso/pull/88913

## Proposed Changes

Adds a new `/switch-site` page which displays the users sites and then after the user selects one, redirects to that site with the given `route` param (e.g. `/switch-site?route=/wp-admin/edit.php` or `/switch-site?route=/posts/:site`). 

Screenshot 2024-04-05 at 16 28 23

### Issues & potential issues
For any given page, the user may or may not have access to that page on some or all of their other sites. Due to plugins, we can't know this in advance for atomic sites. _Maybe_ something could be worked out for simple sites but I'm not sure it's worth the effort. This means that all sites are listed in the site selector and we rely upon the site/plugin to handle this reasonably. Here's what it looks like for the plugins page:

https://github.com/Automattic/wp-calypso/assets/1233880/f8e20979-7631-4d00-8592-58ab8be24524

## Testing Instructions

- Use the Calypso live link below
- Go to `/switch-site?route=<PATH>` and try different routes (e.g. /wp-admin/edit.php, or /posts/:site)
- You should see a list of your sites 
- Choose a site
- You should get sent to the given route for that site

### CODE REVIEW

**Comment by dsas** [comment]
The site selection page currently looks like this:

Screenshot 2024-03-27 at 18 54 43

I'm sure it could look better

**Comment by javierarce** [comment]
Could we do something like this? So, basically a command palette without the shadows. And the left chevron would allow the user to go back to the previous screen.

image

**Comment by mmtr** [comment]
After unifying the commands in https://github.com/Automattic/wp-calypso/pull/89040, we cannot really add the command first to Calypso and then to WP Admin, it needs to be done at the same time. I'll re-work these PRs.

**Comment by mmtr** [comment]
Pushed an update that re-uses the command palette with some modifications to use it as a site selector:

Screenshot 2024-04-05 at 16 28 23

@javierarce let me know if this is what you had in mind. You can also check it using the Calypso live link above and going to `/switch-site?route=%2Fwp-admin%2Fedit.php`

**Review by xavier-lc — APPROVED** [review]
Works as described 👌

**Comment by mmtr** [comment]
> Before this is merged we need to have systems add a routing rule

AFAICS, we first need to ship the new page, and then make the systems requests

**Comment by mmtr** [comment]
> > Before this is merged we need to have systems add a routing rule
> 
> AFAICS, we first need to ship the new page, and then make the systems requests

Moved that requirement to https://github.com/Automattic/wp-calypso/pull/88913

File: client/switch-site/components/site-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			} else {
+				params.navigate( `/home/${ params.site.slug }` );
+			}
+		},
+	};
+
+	return (
+		<main>
+			<DocumentHead title={ __( 'Choose site' ) } />
+			<CommandPalette
```
Comment (dsas) [inline client/switch-site/components/site-switch.tsx:34]: nice!

**Comment by dsas** [comment]
The new changes LGTM, thank you @mmtr

**Pull request merged**